src/Microsoft.DotNet.Darc/src/DarcLib/LocalGitClient.cs

```diff
@@ -149,46 +149,60 @@ public Task MergePullRequestAsync(string pullRequestUrl, MergePullRequestParamet
-                                default:
-                                    throw new DarcException($"Unknown file content encoding {file.ContentEncoding}");
-                            }
-                            finalContent = NormalizeLineEndings(fullPath, finalContent);
-                            await streamWriter.WriteAsync(finalContent);
-                        }
-                        break;
-                    case GitFileOperation.Delete:
-                        if (File.Exists(file.FilePath))
-                        {
-                            File.Delete(file.FilePath);
+                                if (!Directory.Exists(parentDirectory))
+                                {
+                                    Directory.CreateDirectory(parentDirectory);
+                                }
+
+                                string fullPath = Path.Combine(repoUri, file.FilePath);
+                                using (var streamWriter = new StreamWriter(fullPath))
+                                {
+                                    string finalContent;
+                                    switch (file.ContentEncoding)
+                                    {
+                                        case ContentEncoding.Utf8:
+                                            finalContent = file.Content;
+                                            break;
+                                        case ContentEncoding.Base64:
+                                            byte[] bytes = Convert.FromBase64String(file.Content);
+                                            finalContent = Encoding.UTF8.GetString(bytes);
+                                            break;
+                                        default:
+                                            throw new DarcException($"Unknown file content encoding {file.ContentEncoding}");
+                                    }
+                                    finalContent = NormalizeLineEndings(fullPath, finalContent);
+                                    await streamWriter.WriteAsync(finalContent);
+
+                                    LibGit2SharpHelpers.AddFileToIndex(localRepo, file, fullPath, _logger);
+                                }
+                                break;
+                            case GitFileOperation.Delete:
+                                if (File.Exists(file.FilePath))
+                                {
+                                    File.Delete(file.FilePath);
+                                }
+                                break;
                         }
-                        break;
+                    }
+                    LibGit2Sharp.Commands.Stage(localRepo, filesToCommit.Select(f => f.FilePath));
                 }
             }
+            catch (Exception exc)
+            {
+                throw new Exception($"Something went wrong when checking out {repoUri} in {repoDir}", exc);
```

Comment:
I think this should probably be a DarcException (IIRC that's our standard wrapper)